sdks/python/setup.py

```diff
@@ -113,11 +113,8 @@ def get_version():
     'futures>=3.1.1,<4.0.0',
     ]
 
-REQUIRED_SETUP_PACKAGES = [
-    'nose>=1.0',
```

Comment:
Is not this required by setup, because test_suite references it?

Reply:
Does it make sense to raise __REQUIRED_PIP_VERSION = '7.0.0'__ to something a closer to the current 9.0.1 or do we leave that alone for now?

Reply:
Is there a reason for us to raise the required version?

> Alternatively, should the process do pip install --upgrade pip?
Which process? As far as I know, virtualenv brings a new version of pip with it.

Reply:
Python packaging has changed a lot in the past three years especially in relation to Python 3 requirements https://pip.pypa.io/en/stable/news  We want to avoid compatibility issues in the migration to https://github.com/pypa/warehouse

Reply:
Python packaging has changed a lot in the past three years especially in relation to Python 3 requirements https://pip.pypa.io/en/stable/news/

The virtualenv does bring a pip but it can be upgraded by pip (bootstrapping).

Reply:
@cclauss Are you referring to a specific issue? `REQUIRED_PIP_VERSION` is just the minimum version we require, we are not force downgrading to that version. If there is a known issue in compatibility we can consider raising the minimum supported version

Reply:
@udim Is it required to run `python setup.py test`. If not we can drop it.